spec/models/service_provider_spec.rb

```diff
@@ -28,6 +28,112 @@
     end
   end
 
+  describe 'Validations' do
```

Comment:
Do we need all of these tests, or is this something we could test in the gem?

Reply:
@jmhooper: to test in the gem, I'd have to either bring in activerecord and set up a test database, or create a nest of mocks that then seems less than useful.  I decided that, since the apps importing the gem could (should?) validate their requirements, I would host the tests with the apps. Note, this only adds ~1.5 secs to the runtime of the tests. This also documents the new validation requirements in the IDP.

That said, I am running the same tests in the dashboard, so if you prefer not having them in the IDP as well, I'm ok with that.

Reply:
No totally makes that you don't want to bring in AR

Reply:
@jmhooper if you have recommendations for future iterations on our spec strategy for this gem, I'm all ears. I'd even consider bringing in AR to the gem in a future iteration to centralize specs.